src/parcsr_ls/ads.c

```diff
@@ -1609,7 +1609,7 @@ HYPRE_Int hypre_ADSSolve(void *solver,
          r_norm = hypre_sqrt(hypre_ParVectorInnerProd(ads_data -> r0, ads_data -> r0));
          r0_norm = r_norm;
          b_norm = hypre_sqrt(hypre_ParVectorInnerProd(b, b));
-         if (b_norm)
+         if (b_norm != 0.0)
```

Comment:
We should probably use HYPRE_REAL_EPSILON for comparisons against zero where the value is computed. -->  if (b_norm > HYPRE_REAL_EPSILON) ...

Reply:
Sounds great, Daniel! Let's revisit this for sure. I agree using `HYPRE_REAL_MIN` will be more appropriate than `!= 0.0` in some parts of the code...